lib/chatbox-utils.c

```diff
@@ -80,3 +82,421 @@ chatbox_utils_copy_file_to_clipboard (GtkWidget *widget,
+}
+
+static gboolean
+chatbox_key_file_path_equal (const ChatboxKeyFilePath *a,
+                             const ChatboxKeyFilePath *b)
+{
+  return (g_str_equal (a->group_name, b->group_name) &&
+          g_str_equal (a->key_name, b->key_name));
+}
+
+static gboolean
+rewrite_desktop_file_for_flatpak_friendliness (GFile         *source,
+                                               GPtrArray     *key_file_paths,
+                                               GFile         *dest,
+                                               GCancellable  *cancellable,
+                                               GError       **error)
+{
+  g_autoptr(GFile) parent = g_file_get_parent (dest);
+  g_autoptr(GHashTable) key_file_paths_set =
+    ptr_array_to_hash_table_scoped (key_file_paths,
+                                    (GHashFunc) chatbox_key_file_path_hash,
+                                    (GEqualFunc) chatbox_key_file_path_equal);
+  g_autoptr(GError) local_error = NULL;
+  g_autofree char *source_path = g_file_get_path (source);
+  g_autofree char *dest_path = g_file_get_path (dest);
+  g_autoptr(GKeyFile) keyfile = g_key_file_new ();
+  g_auto(GStrv) groups = NULL;
+  gchar **groupp = NULL;
+
+  /* First, make the parent directory if it does not already exist */
+  if (!g_file_make_directory_with_parents (parent, cancellable, &local_error))
+    {
+      if (!g_error_matches (local_error, G_IO_ERROR, G_IO_ERROR_EXISTS))
+        {
+          g_propagate_error (error, g_steal_pointer (&local_error));
+          return FALSE;
+        }
+      
+      g_clear_error (&local_error);
+    }
+
+  /* Read the keyfile and iterate the groups and keys */
+  if (!g_key_file_load_from_file (keyfile, source_path, G_KEY_FILE_NONE, error))
+    return FALSE;
+
+  groups = g_key_file_get_groups (keyfile, NULL);
+
+  for (groupp = groups; *groupp != NULL; ++groupp)
+    {
+      g_auto(GStrv) keys = g_key_file_get_keys (keyfile,
+                                                *groupp,
+                                                NULL,
+                                                error);
+      gchar **keyp = NULL;
+
+      if (keys == NULL)
+        return FALSE;
+
+      for (keyp = keys; *keyp != NULL; ++keyp)
+        {
+          ChatboxKeyFilePath path = {
+            *groupp,
+            *keyp
+          };
+
+          if (g_hash_table_contains (key_file_paths_set, &path))
+            {
+              g_autofree char *path = g_key_file_get_string (keyfile,
+                                                             *groupp,
+                                                             *keyp,
+                                                             error);
+              g_autofree char *fixed_path = NULL;
+
+              if (path == NULL)
+                return FALSE;
+
+              fixed_path = g_build_filename ("/var/run/host", path, NULL);
+              g_key_file_set_string (keyfile,
```

Comment:
Hmmmm ... maybe trying to exec a binary on the host isnt going to end so well if we dont also end up preloading the host libraries. At the very least we should be using flatpak-spawn for this... or remove this hack alltogether if possible.